File: src/frontend/src/app/Entities/Project.ts
export interface Project {
  id: number;
  projectName: string;
  description?: string;
  startDate: Date;
  endDate: Date;
  projectStatus: ProjectStatus;
  progress: number;
}

export enum ProjectStatus {
  Proposed,
  InProgress,
  Completed,
  Archived,
}

File: src/backend/Entities/Project.cs
namespace backend.Entities
{
    public enum ProjectStatus
    {
        Proposed,
        InProgress,
        Completed,
        Archived
    }

    public class Project
    {
        public int Id { get; set; }
        public string ProjectName { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int Progress { get; set; } = 0;
        public ProjectStatus ProjectStatus { get; set; }
    }
}